File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/Core/Scale.java
package com.spvessel.spacevil.Core;

/**
 * Scale is a class that describes the scaling factors along the X and Y axes.
 */
public class Scale {

    private float _x = 1f;
    private float _y = 1f;

    /**
     * Default Scale constructor. All scaling factors are 1.
     */
    public Scale() {
    }

    /**
     * Constructs Scale with specified scaling factors.
     * @param xScale Scaling factor by X axis.
     * @param yScale Scaling factor by Y axis.
     */
    public Scale(float xScale, float yScale) {
        setScale(xScale, yScale);
    }

    /**
     * Setting scaling factors.
     * @param xScale Scaling factor by X axis.
     * @param yScale Scaling factor by Y axis.
     */
    public void setScale(float xScale, float yScale) {
        if (xScale <= 0 || yScale <= 0) {
            return;
        }
        _x = xScale;
        _y = yScale;
    }

    /**
     * Getting scaling factor by X axis.
     * @return Scaling factor by X axis.
     */
    public float getXScale() {
        return _x;
    }

    /**
     * Getting scaling factor by Y axis.
     * @return Scaling factor by Y axis.
     */
    public float getYScale() {
        return _y;
    }

    @Override
    public String toString()
    {
        return "XScale: " + _x + " YScale: " + _y;
    }
}

File: CSharp/SpaceVIL/SpaceVIL/Core/Scale.cs
using System;

namespace SpaceVIL.Core
{
    /// <summary>
    /// Scale is a class that describes the scaling factors along the X and Y axes.
    /// </summary>
    public class Scale
    {
        private float _x = 1f;
        private float _y = 1f;
        /// <summary>
        /// Default Scale constructor. All scaling factors are 1.
        /// </summary>
        public Scale() { }
        /// <summary>
        /// Constructs Scale with specified scaling factors.
        /// </summary>
        /// <param name="xScale">Scaling factor by X axis.</param>
        /// <param name="yScale">Scaling factor by Y axis.</param>
        public Scale(float xScale, float yScale)
        {
            SetScale(xScale, yScale);
        }
        /// <summary>
        /// Setting scaling factors.
        /// </summary>
        /// <param name="xScale">Scaling factor by X axis.</param>
        /// <param name="yScale">Scaling factor by Y axis.</param>
        public void SetScale(float xScale, float yScale)
        {
            if (xScale <= 0 || yScale <= 0)
                return;
            _x = xScale;
            _y = yScale;
        }
        /// <summary>
        /// Getting scaling factor by X axis.
        /// </summary>
        /// <returns>Scaling factor by X axis.</returns>
        public float GetXScale()
        {
            return _x;
        }
        /// <summary>
        /// Getting scaling factor by Y axis.
        /// </summary>
        /// <returns>Scaling factor by Y axis.</returns>
        public float GetYScale()
        {
            return _y;
        }

        public override string ToString()
        {
            return "XScale: " + _x.ToString(".0##") + " YScale: " + _y.ToString(".0##");
        }
    }
}